app/src/debug/kotlin/uk/gov/onelogin/appcheck/AppCheckModule.kt

```diff
@@ -19,22 +20,22 @@ object AppCheckModule {
     fun provideAppCheck(
         @ApplicationContext
         context: Context
-    ): AppCheck = FirebaseAppCheck().also {
-        val key = Firebase.app.persistenceKey
-        context.getSharedPreferences(
-            "com.google.firebase.appcheck.debug.store.$key",
-            MODE_PRIVATE
-        ).edit().apply {
-            putString(
-                "com.google.firebase.appcheck.debug.DEBUG_SECRET",
-                BuildConfig.AppCheckDebugSecret
-            )
-            commit()
-        }
+    ): AppChecker {
+        val factory = DebugAppCheckProviderFactory.getInstance()
+        return FirebaseAppCheck(factory).also {
+            val key = Firebase.app.persistenceKey
+            context.getSharedPreferences(
+                "com.google.firebase.appcheck.debug.store.$key",
+                MODE_PRIVATE
+            ).edit().apply {
```

Comment:
Please use the SharedPreferences kotlin extensions `sharedPrefs.edit(true) {}`, because we are using them elsewhere so it is consistent.
Also I think it would be best to pull the SharedPreferences call out into a separate function, having it here means we don't need to test it for coverage, but we should test as much as possible.

I will leave the decision on these up to you, do not feel obliged to do these